drivers/wifi/esp_at/esp.c

```diff
@@ -1166,6 +1160,14 @@ static int esp_init(const struct device *dev)
 			K_NO_WAIT);
 	k_thread_name_set(&esp_rx_thread, "esp_rx");
 
+	/* Retrieve associated network interface so asynchronous messages can be processed early */
+	data->net_iface = NET_IF_GET(Z_DEVICE_DT_DEV_NAME(DT_DRV_INST(0)), 0);
+
+	/* Reset the modem */
+	net_if_flag_set(data->net_iface, NET_IF_NO_AUTO_START);
+	esp_offload_init(data->net_iface);
+	ret = esp_reset(data);
```

Comment:
I like the fact that `esp_reset()` is called from `esp_init()` instead of `esp_iface_init()`, it is more logical as resettting device is not "network interface" related.

I wonder if `esp_offload_init()` and `net_if_flag_set()` called from `esp_init()` is a good change, though.

Reply:
@JordanYates ^^

Reply:
I was originally concerned that the ready callback would rely on those values being set somehow, but I have convinced myself it doesn't. I have put those calls back in `esp_iface_init`.